### PR TITLE
Adds unread changes notification

### PR DESCRIPTION
**Pull request by CrAzYPiLoT-SS13**
As title. When there are new changes which the client hasn't already read, the changelog button will bold  and become reddish to note that you should check the changelog. Also, a note is displayed in the chat. 
screenshot_28
Note, this _should_ work, but I have no way of telling that because I don't have a SQL DB set up. I have checked for errors and other stuff to the extent I could. As such, maintainers, please review this thoroughly. 
:rotating_light: SQL change needed! :rotating_light: 
```
ALTER TABLE `player`
ADD `lastchangelog` varchar(32) NOT NULL
```

:cl: CrAzYPiLoT
rscadd: Adds an unread change notification.
/:cl:

### CODE REVIEW

**Comment by Regen1** [comment]
@TheDZD this is your field I think

**Comment by pinatacolada** [comment]
from a ux point of view, people won't notice it, but that's due to the nature of the location of the changelog button. The karma shop button having a similarish color doesn't help it either. just my 2cents, I like the function itself

**Comment by CrAzYPiLoT-SS13** [comment]
Uh, there is a message saying "Changelog has changed since your last visit.". It's visible on the screenie.

**Comment by pinatacolada** [comment]
see what I mean with visible message? I didn't even notice it was there, it needs to be a larger font at the very least

**Comment by Spacemanspark** [comment]
Size 72 font yes.

**Comment by CrAzYPiLoT-SS13** [comment]
This is what is used for info messages. I would count this into info, wouldn't you?

File: code/modules/client/preferences_mysql.dm [hunk 1]
```diff
@@ -75,7 +77,8 @@
 					randomslot='[randomslot]',
 					volume='[volume]',
 					nanoui_fancy='[nanoui_fancy]',
-					show_ghostitem_attack='[show_ghostitem_attack]'
+					show_ghostitem_attack='[show_ghostitem_attack]',
+					lastchangelog='[lastchangelog]',
```
Comment (tigercat2000) [inline code/modules/client/preferences_mysql.dm]: trailing comma bad

File: code/modules/client/preferences_mysql.dm [hunk 2]
```diff
@@ -449,4 +452,14 @@
 		load_character(C)
 		return 0
 	load_character(C,pick(saves))
-	return 1*/
\ No newline at end of file
+	return 1*/
+
+/datum/preferences/proc/SetChangelog(client/C,hash)
+	lastchangelog=hash
+	var/DBQuery/query = dbcon.NewQuery("UPDATE client SET lastchangelog='[lastchangelog]' WHERE ckey='[C.ckey]'")
```
Comment (tigercat2000) [inline code/modules/client/preferences_mysql.dm]: this won't work, replace it with `UPDATE [format_table_name("player")] SET lastchangelog='[lastchangelog]' WHERE ckey='[C.ckey]'`

**Comment by KasparoVy** [comment]
Seems fine to me, nice feature

File: code/modules/client/client procs.dm [hunk 3]
```diff
@@ -308,6 +308,11 @@
 
 	send_resources()
 
+	if(prefs.lastchangelog != changelog_hash) //bolds the changelog button on the interface so we know there are updates. -CP
+		winset(src, "rpane.changelog", "background-color=#f4aa94;font-style=bold")
+		prefs.SetChangelog(src,changelog_hash)
+		src << "<span class='info'>Changelog has changed since your last visit.</span>"
```
Comment (PJB3005) [inline code/modules/client/client procs.dm]: `to_chat`

File: code/modules/client/preferences_mysql.dm (hunk 1 above)
Comment (monster860) [inline code/modules/client/preferences_mysql.dm]: IMO, we should have trailing commas. They keep diffs small.

Comment (tigercat2000) [inline code/modules/client/preferences_mysql.dm]: they're also invalid SQL syntax and would make this query not work

**Comment by Iamgoofball** [comment]
just open the changelog if they havent seen the latest changelog

**Comment by CrAzYPiLoT-SS13** [comment]
I can do that, if it's needed, I didn't want to force something on them,
like those annoying polls some servers still have.

**Comment by Iamgoofball** [comment]
force it on them anyways

you do not have the right to be ignorant

**Comment by RemieRichards** [comment]
at TG we've forced it open on un-viewed changes for ages, like, atleast a year+, it's good.

**Comment by CrAzYPiLoT-SS13** [comment]
I don't know. If the community wants it, I'll add it.

**Comment by CrAzYPiLoT-SS13** [comment]
What's the status on this? I do not plan on forcing the CL open on join, unless one of the maintainers requests it.

**Comment by CrAzYPiLoT-SS13** [comment]
I don't even know what's happening with the maintainers, but they seem to not pay attention to any of the open non-port PRs. I have other things in mind, and I would like to make PRs for them, but I would first rather know if this will even be discussed, let alone merged.

**Comment by Fox-McCloud** [comment]
>but they seem to not pay attention to any of the open non-port PRs.

Apologies for having a life outside of SS13. 

Secondly, this has a SQL change tied to it and *any* PR that has a SQL change tied to it cannot be merge in until we patch--that opportunity hasn't come up yet, which is why this hasn't been merged in.

**Comment by CrAzYPiLoT-SS13** [comment]
I do not blame you for having a life, sorry if you got that feeling from my
comment. But, I /have/ seen you active on other PRs and that's what was my
problem. Sorry again if you got offended.

**Comment by CrAzYPiLoT-SS13** [comment]
Can I at least get an approximation of the patch date?

**Comment by CrAzYPiLoT-SS13** [comment]
I guess I can't.

**Comment by tigercat2000** [comment]
we don't patch on a schedule
it's literally just "when the right people aren't busy"

**Comment by PJB3005** [comment]
Might wanna consider stealing /vg/'s migrations to remove the headache of patching the SQL.

**Comment by TheDZD** [comment]
This was *supposed* to have been merged before the last patch, but it wasn't and I need to whack somebody as a result.

**Comment by CrAzYPiLoT-SS13** [comment]
Spend donation money on DZD's anger management :P

**Comment by CrAzYPiLoT-SS13** [comment]
And another patch passes...

**Comment by Fox-McCloud** [comment]
Neca was directly contacted about the SQL changes right before the patch; he was unsure of how to do them, which is why this didn't get merged in.

**Comment by CrAzYPiLoT-SS13** [comment]
Is my SQL query in the OP not correct? I sure hope it is

**Comment by TheDZD** [comment]
It is correct.

**Comment by tigercat2000** [comment]
well
technically speaking
if neca just blindly posted it in with nothing else setup it wouldn't work
no database target
but setting the default database or including a USE statement would work

**Pull request merged**